pvanalytics/quality/gaps.py

```diff
@@ -135,3 +136,206 @@ def interpolation_diff(x, window=3, rtol=1e-5, atol=1e-8):
         atol=atol
     )
     return flags
+
+
+def _freq_to_seconds(freq):
+    if not freq:
+        return None
+    if freq.isalpha():
+        freq = '1' + freq
+    delta = pd.to_timedelta(freq)
+    return delta.days * (1440 * 60) + delta.seconds
+
+
+def daily_completeness(series, freq=None):
+    """Calculate a completeness score for each day.
+
+    The completeness score for a given day is the fraction of time in
+    the day for which there is data (a value other than NaN). The time
+    attributed to each value is equal to the timestamp spacing of
+    `series` or `freq` if it is specified. For example, a day with 24
+    non-NaN values in a series with 30 minute timestamp spacing would
+    have 12 hours of data and therefore completeness score of 0.5.
+
+    Parameters
+    ----------
+    series : Series
+        A DatetimeIndexed series.
+    freq : str, default None
+        Interval between samples in the series as a pandas frequency
+        string. If None, the frequency is inferred using
+        :py:func:`pandas.infer_freq`.
+
+    Returns
+    -------
+    Series
+        A series of floats, indexed by day, giving the completeness
+        score for each day (fraction of the day for which `series` has
+        data).
+
+    Raises
+    ------
+    ValueError
+        If `freq` is longer than the frequency inferred from `series`.
+
+    """
+    inferred_seconds = _freq_to_seconds(pd.infer_freq(series.index))
+    freq_seconds = _freq_to_seconds(freq)
+    seconds_per_sample = freq_seconds or inferred_seconds
```

Comment:
The `or` operator short-circuits, returning `freq_seconds` in that case. I think it isn't clear because as you read the code you assume `_freq_to_seconds` always returns a value (if its argument is `None` then it returns `None`). I will re-work it for clarity.

Reply:
Or put some inline comments for future maintainers

Reply:
I ended up rewriting it. I think there is just a little too much going on for the `arg or default` pattern to be clear.